yt/yt/core/http/retrying_client.cpp

```diff
@@ -18,14 +18,44 @@ static const auto& Logger = HttpLogger;
 
 ////////////////////////////////////////////////////////////////////////////////
 
-class TJsonResponseChecker
+class TResponseCheckerBase
     : public IResponseChecker
 {
+public:
+    TResponseCheckerBase(TRetryChecker retryChecker = {})
+        : RetryChecker_(retryChecker ? std::move(retryChecker) : BIND(&DefaultRetryChecker))
+    { }
+
+    virtual bool IsRetriableError(const TError& error) override 
+    {
+        return RetryChecker_(error);
+    }
+
+    virtual TError CheckError(const IResponsePtr& response) override = 0;
+    virtual NYTree::INodePtr GetFormattedResponse() const override = 0;
+
+protected:
+    static bool DefaultRetryChecker(const TError& error)
+    {
+        return !error.FindMatching(NRpc::EErrorCode::InvalidCredentials).has_value();
```

Comment:
It looks a little bit weird in the base class, I suppose that here we should just `return true`

But in oauth_service.cpp we can specify checker that make retries for all errors expect InvalidCredentials